python/ray/tune/execution/tune_controller.py

```diff
@@ -2029,6 +2154,16 @@ def cleanup(self):
         self._cleanup_trials()
         self.end_experiment_callbacks()
 
+        # Upload train logdir in /tmp/ray/session_*/train
+        from ray.train._internal.storage import _upload_to_fs_path, get_train_logdir
+
+        logger.info(f"Copying logs from {get_train_logdir()} to storage...")
+        _upload_to_fs_path(
+            get_train_logdir(),
+            fs=self._storage.storage_filesystem,
+            fs_path=self._storage.experiment_fs_path,
+        )
+
```

Comment:
Change 3: `error.txt` and `error.pkl` are 2 other driver artifacts that get generated. Store these in the ray log directory and upload them at the end of the experiment.

The driver writes `error.txt` / `error.pkl` files into the `/tmp/ray/session_*/train/` folder, and we upload this dir on experiment end.

The benefit of this folder is that you can view it in Ray dashboard, it's together with other ray logs, and users can actually find it (compared to random tempdirs that we make).